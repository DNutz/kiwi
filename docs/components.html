<!DOCTYPE html>
<!--[if IE 8]> <html lang="en" class="ie8"> <![endif]-->  
<!--[if IE 9]> <html lang="en" class="ie9"> <![endif]-->  
<!--[if !IE]><!--> <html lang="en"> <!--<![endif]-->  
<head>
    <title>Responsive website template for documentations</title>
    <!-- Meta -->
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="">
    <meta name="author" content="">    
    <link rel="shortcut icon" href="favicon.ico">  
    <link href='http://fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,800italic,400,300,600,700,800' rel='stylesheet' type='text/css'>
    <!-- Global CSS -->
    <link rel="stylesheet" href="assets/plugins/bootstrap/css/bootstrap.min.css">   
    <!-- Plugins CSS -->    
    <link rel="stylesheet" href="assets/plugins/font-awesome/css/font-awesome.css">
    <link rel="stylesheet" href="assets/plugins/prism/prism.css">
    <link rel="stylesheet" href="assets/plugins/lightbox/dist/ekko-lightbox.min.css">
    <link rel="stylesheet" href="assets/plugins/elegant_font/css/style.css">

    <!-- Theme CSS -->
    <link id="theme-style" rel="stylesheet" href="assets/css/styles.css">
    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
    
</head> 

<body class="body-pink">
    <div class="page-wrapper">
        <!-- ******Header****** -->
        <header id="header" class="header">
            <div class="container">
                <div class="branding">
                    <h1 class="logo">
                        <a href="index.html">
                            <span aria-hidden="true" class="icon_documents_alt icon"></span>
                            <span class="text-highlight">Team</span><span class="text-bold">Kiwi</span>
                        </a>
                    </h1>
                </div><!--//branding-->
                <ol class="breadcrumb">
                    <li><a href="index.html">Home</a></li>
                    <li class="active">Components</li>
                </ol>
            </div><!--//container-->
        </header><!--//header-->
        <div class="doc-wrapper">
            <div class="container">
                <div id="doc-header" class="doc-header text-center">
                    <h1 class="doc-title"><span aria-hidden="true" class="icon icon_puzzle_alt"></span> Components</h1>
                    <div class="meta"><i class="fa fa-clock-o"></i> Last updated: May 30th, 2018</div>
                </div><!--//doc-header-->
                <div class="doc-body">
                    <div class="doc-content">
                        <div class="content-inner">
                            <section id="login" class="doc-section">
                                <h2 class="section-title">Login Screen</h2>
                                <div class="section-block">
                                        <div>
                                                <h6>Main Login Page</h6>
                                                <p></p>
                                                <div class="screenshot-holder">
                                                    <a href="assets/images/login.PNG" data-title="Login" data-toggle="lightbox"><img class="img-responsive" src="assets/images/login.PNG" alt="screenshot" /></a>
                                                    <a class="mask" href="assets/images/login.PNG" data-title="Login" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                                </div> 
                                            </div>
                                
                                    <div id="Passwordless" class="row">
                                        <h3 class="block-title">Passwordless Login</h3>
                                        <div class="col-md-6 col-sm-12 col-sm-12">
                                            <h6>How it looks</h6>
                                            <p></p>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/PasswordlessLook.PNG" data-title="Login" data-toggle="lightbox"><img class="img-responsive" src="assets/images/PasswordlessLook.PNG" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/PasswordlessLook.PNG" data-title="Login" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-6 col-sm-12 col-sm-12">
                                                <h6>Code</h6>
                                                <div class="screenshot-holder">
                                                    <a href="assets/images/Passwordless.PNG" data-title="Passwordless" data-toggle="lightbox"><img class="img-responsive" src="assets/images/Passwordless.PNG" alt="screenshot" /></a>
                                                    <a class="mask" href="assets/images/Passwordless.PNG" data-title="Passwordless" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                                </div> 
                                            </div>
                                    </div><!--//row-->
                                </div><!--//section-block-->
                                
                            </section><!--//doc-section-->
                            <section id="app-components" class="doc-section">
                                <h2 class="section-title">App Components</h2>
                                <div id="projects" class="section-block">
                                    <h3 class="block-title">Projects</h3>
                                    <p>Intro goes here lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis</p>
                                    <div class="row">
                                        <div class="col-md-6 col-sm-6 col-xs-12">
                                            <h6>Projects Overview</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-projects.jpg" data-title="Project Overview" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-projects-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-projects.jpg" data-title="Project Overview" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-6 col-sm-6 col-xs-12">
                                            <h6>Project Details</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-project.jpg" data-title="Project Details" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-project-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-project.jpg" data-title="Project Details" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                    </div><!--//row-->
                                </div><!--//section-block-->
                                <div id="discussions" class="section-block">
                                    <h3 class="block-title">Discussions</h3>
                                    <div class="row">
                                        <div class="col-md-6 col-sm-6 col-xs-12">
                                            <h6>Discussions Overview</h6>
                                            <p>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes.</p>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-discussions.jpg" data-title="Discussion Overview" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-discussions-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-discussions.jpg" data-title="Discussion Overview" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-6 col-sm-6 col-xs-12">
                                            <h6>Discussion Details</h6>
                                            <p>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes.</p>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-discussion.jpg" data-title="Discussion Details" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-discussion-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-discussion.jpg" data-title="Discussion Details" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                    </div><!--//row-->  
                                </div><!--//section-block-->
                                
                                <div id="members" class="section-block">
                                    <h3 class="block-title">Members</h3>
                                    <div class="row">
                                        <div class="col-md-6 col-sm-6 col-xs-12">
                                            <h6>Members Overview</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-members.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-members-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-members.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-6 col-sm-6 col-xs-12">
                                            <h6>Individual Member</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-member.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-member-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-member.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                    </div><!--//row-->  
                                </div><!--//section-block-->
                                
                                <div id="account" class="section-block">
                                    <h3 class="block-title">User Account</h3>
                                    <p>Intro goes here lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes.</p>
                                    <div class="row">
                                        <div class="col-md-4 col-sm-4 col-xs-6">
                                            <h6>User Profile</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-account-profile.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-account-profile-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-account-profile.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-4 col-sm-4 col-xs-6">
                                            <h6>User Settings</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-account-settings.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-account-settings-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-account-settings.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-4 col-sm-4 col-xs-6">
                                            <h6>User Billing</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-account-billing.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-account-billing-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-account-billing.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-4 col-sm-4 col-xs-6">
                                            <h6>User Messages</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-account-messages.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-account-messages-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-account-messages.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-4 col-sm-4 col-xs-6">
                                            <h6>User Drive</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-account-drive.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-account-drive-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-account-drive.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-4 col-sm-4 col-xs-6">
                                            <h6>User Reminders</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-account-reminders.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-account-reminders-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-account-reminders.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                    </div><!--//row-->  
                                </div><!--//section-block-->
                                
                                 <div id="others" class="section-block">
                                    <h3 class="block-title">Others</h3>
                                    <div class="row">
                                        <div class="col-md-3 col-sm-3 col-xs-6">
                                            <h6>Activities</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-activities.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-activities-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-activities.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-3 col-sm-3 col-xs-6">
                                            <h6>Tickets</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-tickets.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-tickets-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-tickets.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-3 col-sm-3 col-xs-6">
                                            <h6>Inbox</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-inbox.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-inbox-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-inbox.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                        <div class="col-md-3 col-sm-3 col-xs-6">
                                            <h6>Timeline</h6>
                                            <div class="screenshot-holder">
                                                <a href="assets/images/demo/appkit-timeline.jpg" data-toggle="lightbox"><img class="img-responsive" src="assets/images/demo/appkit-timeline-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="assets/images/demo/appkit-timeline.jpg" data-toggle="lightbox"><i class="icon fa fa-search-plus"></i></a>
                                            </div> 
                                        </div>
                                    </div><!--//row-->  
                                </div><!--//section-block-->
                                <div class="section-block">
                                    <p class="text-center">Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes.</p>
                                    <div class="center-block text-center">
                                         <a class="btn btn-primary" href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank">More App Components</a>
                                    </div>
                                </div><!--//section-block-->
                            </section><!--//doc-section-->
                            <section id="ui-components" class="doc-section">
                                <h2 class="section-title">UI Components</h2>
                                <div class="section-block">
                                    <p>Appkit is packed with building blocks. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. </p>
                                    <div class="row">
                                        <div class="col-md-4 col-sm-5 col-xs-12">
                                            <h6>Components List</h6>
                                            <ul class="list list-unstyled">
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Typography</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Icons</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Buttons</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Panels</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Tabs</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Accordions</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Maps</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Todo</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Carousels</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Notification</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Draggable</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Sortable</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Nestable</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Tree</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Loaders</a></li>
                                                <li><a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="fa fa-check"></i> Many More...</a></li>
                                            </ul>
                                        </div>
                                        <div class="col-md-8 col-sm-5 col-xs-12">
                                            <h6>Screenshot</h6>
                                            <div class="screenshot-holder">
                                                <a href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><img class="img-responsive" src="assets/images/demo/appkit-widgets-thumb.jpg" alt="screenshot" /></a>
                                                <a class="mask" href="https://wrapbootstrap.com/theme/appkit-admin-theme-angularjs-WB051SCJ1?ref=3wm" target="_blank"><i class="icon fa fa-link"></i></a>
                                                </div> 
                                        </div>
                                    </div><!--//row-->
                                </div><!--//section-block-->
                            </section><!--//doc-section-->
                            
                        </div><!--//content-inner-->
                    </div><!--//doc-content-->
                    <div class="doc-sidebar">
                        <nav id="doc-nav">
                            <ul id="doc-menu" class="nav doc-menu hidden-xs" data-spy="affix">
                                <li>
                                    <a class="scrollto" href="#login">Login Screen</a>
                                    <ul class="nav doc-sub-menu">
                                        <li><a class="scrollto" href="#Passwordless">Passwordless Login</a></li>
                                    </ul><!--//nav-->
                                </li>
                                <li>
                                    <a class="scrollto" href="#app-components">App Components</a>
                                    <ul class="nav doc-sub-menu">
                                        <li><a class="scrollto" href="#projects">Projects</a></li>
                                        <li><a class="scrollto" href="#discussions">Discussions</a></li>
                                        <li><a class="scrollto" href="#members">Members</a></li>
                                        <li><a class="scrollto" href="#account">User Account</a></li>
                                        <li><a class="scrollto" href="#others">Others</a></li>
                                    </ul><!--//nav-->
                                </li>
                                <li><a class="scrollto" href="#ui-components">UI Components</a></li>
                            </ul><!--//doc-menu-->
                        </nav>
                    </div><!--//doc-sidebar-->
                </div><!--//doc-body-->              
            </div><!--//container-->
        </div><!--//doc-wrapper-->
        
        
        
    </div><!--//page-wrapper-->
    
    <footer id="footer" class="footer text-center">
        <div class="container">
            <!--/* This template is released under the Creative Commons Attribution 3.0 License. Please keep the attribution link below when using for your own project. Thank you for your support. :) If you'd like to use the template without the attribution, you can check out other license options via our website: themes.3rdwavemedia.com */-->
            <small class="copyright">Website Design Designed with <i class="fa fa-heart"></i> by <a href="http://themes.3rdwavemedia.com/" target="_blank">Xiaoying Riley</a> for developers</small>
            
        </div><!--//container-->
    </footer><!--//footer-->
    
     
    <!-- Main Javascript -->          
    <script type="text/javascript" src="assets/plugins/jquery-1.12.3.min.js"></script>
    <script type="text/javascript" src="assets/plugins/bootstrap/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="assets/plugins/prism/prism.js"></script>    
    <script type="text/javascript" src="assets/plugins/jquery-scrollTo/jquery.scrollTo.min.js"></script>  
    <script type="text/javascript" src="assets/plugins/lightbox/dist/ekko-lightbox.min.js"></script>                                                                 
    <script type="text/javascript" src="assets/plugins/jquery-match-height/jquery.matchHeight-min.js"></script>
    <script type="text/javascript" src="assets/js/main.js"></script>
    
</body>
</html> 

