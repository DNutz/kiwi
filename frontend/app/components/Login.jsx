import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink, withRouter } from 'react-router-dom';
import { push } from 'react-router-redux';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';

// - Import actions
import * as authorizeActions from 'authorizeActions';

export class Login extends Component {

    /**
     * Component constructor
     * @param  {object} props is an object properties of component
     */
    constructor(props) {
        super(props);

        this.state = {
            emailInput: '',
            emailInputError: '',
            passwordInput: '',
            passwordInputError: ''
        };
    }

    /**
     * Handle data on input change
     * @param  {event} evt is an event of inputs of element on change
     */
    handleInputChange = (evt) => {
        const target = evt.target;
        const name = target.name;

        this.setState({ [name]: target.value });

        // Clear text error fields.
        if (name === 'emailInput') {
            this.setState({ emailInputError: '' });
        }

        else if (name === 'passwordInput') {
            this.setState({ passwordInputError: '' });
        }
    }

    handleForm = () => {
        if (this.state.emailInput === '') {
            this.setState({ emailInputError: 'This field is required' });
        }

        else if (this.state.passwordInput === '') {
            this.setState({ passwordInputError: 'This field is required' });
        }

        else {
            this.props.login(this.state.emailInput, this.state.passwordInput);
        }
    }


    /**
     * Reneder component DOM
     * @return {react element} return the DOM which rendered by component
     */
    render() {
        return (
            <form style={{height: '100%', width: '100%', display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
                {/* <div className="animate-bottom"> */}
                    <div style={{backgroundColor: 'white', width: '450px', margin: '20px', textAlign: 'center', display: 'flex', minHeight: '370px', borderRadius: '10px', justifyContent: 'center', alignItems: 'center'}}>
                        <div style={{ padding: "48px 40px 36px" }}>
                            <div style={{paddingLeft: "40px", paddingRight: "40px"}}>
                                <h2 style={{
                                    textAlign: "left",
                                    paddingTop: "16px",
                                    fontSize: "24px",
                                    fontWeight: 400,
                                    lineHeight: "32px",
                                    margin: 0
                                }}>Sign in</h2>
                            </div>

                            <TextField
                                onChange={this.handleInputChange}
                                errorText={this.state.emailInputError}
                                name="emailInput"
                                floatingLabelStyle={{ fontSize: "15px" }}
                                floatingLabelText="Email"
                                type="email"
                                tabIndex={1}
                            /><br />
                            <TextField
                                onChange={this.handleInputChange}
                                errorText={this.state.passwordInputError}
                                name="passwordInput"
                                floatingLabelStyle={{ fontSize: "15px" }}
                                floatingLabelText="Password"
                                type="password"
                                tabIndex={2}
                            /><br />
                            <br />
                            <br />
                            <div className="login__button-box">
                                <div>
                                    <FlatButton label="Create an account" onClick={this.props.signupPage} tabIndex={4} />
                                </div>
                                <div >
                                    <RaisedButton label="Login" primary={true} onClick={this.handleForm} tabIndex={3} />
                                </div>
                            </div>
                        </div>
                    </div>
                {/* </div> */}
            </form>
        )
    }
}

/**
 * Map dispatch to props
 * @param  {func} dispatch is the function to dispatch action to reducers
 * @param  {object} ownProps is the props belong to component
 * @return {object}          props of component
 */
const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        login: (username, password) => {
            dispatch(authorizeActions.dbLogin(username, password))
        },
        signupPage: () => {
            dispatch(push("/signup"))
        }
    }
}

/**
 * Map state to props
 * @param  {object} state is the obeject from redux store
 * @param  {object} ownProps is the props belong to component
 * @return {object}          props of component
 */
const mapStateToProps = (state, ownProps) => {
    return {

    }
}

// - Connect component to redux store
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login))
