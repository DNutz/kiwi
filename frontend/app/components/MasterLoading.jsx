import React from 'react';
import Dialog from 'material-ui/Dialog';

const MasterLoading = ({activeLoading}) => (
    <Dialog
        modal={true}
        open={activeLoading}
        autoDetectWindowHeight={true}
        contentStyle={{width: '250px'}}
    >
        <div style={{display: 'flex', justifyContent: 'space-around'}}>
            <div id="wave">
                <span className="dot"></span>
                <span className="dot"></span>
                <span className="dot"></span>
            </div>
            <h1>Loading</h1>
        </div>
    </Dialog>
);

export default MasterLoading;