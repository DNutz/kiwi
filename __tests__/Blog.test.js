import React from "react";
import expect from "expect";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme"; 
import renderer from "react-test-renderer";
import getMuiTheme from "material-ui/styles/getMuiTheme";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import { Blog } from "../src/app/components/Blog";
import { ListItem } from "material-ui";

Enzyme.configure({ adapter: new Adapter() });

describe("Component: Blog", () => {
    it("renders without crashing", () => {
        const wrapper = shallow(<Blog setHomeTitle={()=>{}} match={{params : ""}}/>);
        expect(wrapper.length).toEqual(1);
    });

    it("renders correctly", () => {
        const component = renderer.create(
            <MuiThemeProvider>
                <Blog setHomeTitle={()=>{}} match={{params: ""}}/>
            </MuiThemeProvider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("has correct default state for openPostWrite", () => {
        const wrapper = Enzyme.mount(<Blog setHomeTitle={()=>{}} match={{params: ""}}/>);
        expect(
            String(wrapper.state("openPostWrite")) === "false"
         ).toEqual(true);
    });

    it("has correct state after handleOpenPostWrite", () => {
        const wrapper = Enzyme.mount(<Blog setHomeTitle={()=>{}} match={{params: ""}}/>);
        // console.log(wrapper.props());
        // wrapper.props().handleOpenPostWrite();
        // wrapper.find('ListItem').simulate('click');
        wrapper.simulate('handleOpenPostWrite');
        expect(
            String(wrapper.state("openPostWrite")) === "true"
         ).toEqual(true);
    });
});

    // it("has correct state after handleOpenPostWrite", () => {
    //     const wrapper = Enzyme.mount(<Blog setHomeTitle={()=>{}} match={{params: ""}}/>);
    //     // console.log(wrapper.props());
    //     // wrapper.props().handleOpenPostWrite();
    //     wrapper.find(ListItem).simulate(TouchTap);
    //     expect(
    //         String(wrapper.state("openPostWrite")) === "true"
    //      ).toEqual(true);
    // });

    // it("receives a proper string in image", () => {
    //     const wrapper = Enzyme.mount(<MuiThemeProvider><Blog/></MuiThemeProvider>);
    //     expect(
    //         typeof(wrapper.childAt(0).prop("image")) === "string"
    //     ).toEqual(true);
    // });

    // it("receives the correct string in image", () => {
    //     const wrapper = Enzyme.mount(<MuiThemeProvider><Blog/></MuiThemeProvider>);
    //     expect(
    //         String(wrapper.childAt(0).prop("image")) === "src/public/images/skyAds/advertise.jpg"
    //     ).toEqual(true);
    // });

    // it("has correct default state type for image", () => {
    //     const wrapper = Enzyme.mount(<MuiThemeProvider><Blog/></MuiThemeProvider>);
    //     expect(
    //         typeof(wrapper.childAt(0).prop("image")) === "undefined"
    //     ).toEqual(true);
    // });

    // it("has correct default state for image", () => {
    //     const wrapper = Enzyme.mount(<MuiThemeProvider><Blog/></MuiThemeProvider>);
    //     expect(
    //         String(wrapper.childAt(0).prop("image")) === "undefined"
    //     ).toEqual(true);
    // });
// });

