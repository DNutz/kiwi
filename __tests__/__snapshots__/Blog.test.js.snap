// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Blog renders correctly 1`] = `
<div>
  
  
  <div
    className="grid grid__gutters grid__1of2 grid__space-around animate-top"
  >
    <div
      className="grid-cell animate-top"
      style={
        Object {
          "maxWidth": "530px",
          "minWidth": "280px",
        }
      }
    >
      
      <h1>
        'Nothing has shared.'
      </h1>
      <div
        style={
          Object {
            "height": "16px",
          }
        }
      />
    </div>
  </div>
</div>
`;
